samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -235,6 +263,7 @@ private JobModel generateNewJobModel(List<String> processors) {
     @Override
     public void onBecomingLeader() {
       LOG.info("ZkJobCoordinator::onBecomeLeader - I became the leader!");
+      metrics.leaderElection.inc();
```

Comment:
@PawasChhokra Looks like this metric counts the number of times a processor has become the leader. The metric name "leader-election" can be easily misinterpreted as the number of times the processor participated in leader election. Can you please rename this metric?